File: ts-sdk/src/sdk/models/operations/getgroups.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetGroupsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=filter_term" })
  filterTerm?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;
}


export class GetGroupsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetGroupsQueryParams;
}


export class GetGroupsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  groups?: shared.Groups;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getgroups.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetGroupsQueryParams struct {
	Fields     []string `queryParam:"style=form,explode=false,name=fields"`
	FilterTerm *string  `queryParam:"style=form,explode=true,name=filter_term"`
	Limit      *int64   `queryParam:"style=form,explode=true,name=limit"`
	Offset     *int64   `queryParam:"style=form,explode=true,name=offset"`
}

type GetGroupsRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetGroupsQueryParams
}

type GetGroupsResponse struct {
	ClientError *shared.ClientError
	ContentType string
	Groups      *shared.Groups
	StatusCode  int64
}
